pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java

```diff
@@ -267,7 +267,7 @@ public PulsarAdminException getApiException(Response response) {
 
     public static String getReasonFromServer(WebApplicationException e) {
         try {
-            return e.getResponse().readEntity(ErrorData.class).reason.toString();
```

Comment:
If we remove the `toString` call here, the `reason` may be null and be returned. We need to make sure that null is not returned here. The correct behavior is to return e.getMessage() when the `reason` is null.